ui/src/shared/components/Dropdown.js

```diff
@@ -164,33 +285,57 @@ class Dropdown extends Component {
       iconName,
       buttonSize,
       buttonColor,
+      useAutoComplete,
     } = this.props
-    const {isOpen} = this.state
+    const {isOpen, searchTerm, filteredItems} = this.state
+    const menuItems = useAutoComplete ? filteredItems : items
 
     return (
       <div
         onClick={this.handleClick}
         className={classnames(`dropdown ${className}`, {open: isOpen})}
       >
-        <div className={`btn dropdown-toggle ${buttonSize} ${buttonColor}`}>
-          {iconName
-            ? <span className={classnames('icon', {[iconName]: true})} />
-            : null}
-          <span className="dropdown-selected">{selected}</span>
-          <span className="caret" />
-        </div>
-        {(isOpen && items.length < DROPDOWN_MENU_ITEM_THRESHOLD)
+        {useAutoComplete && isOpen
+          ? <div
+              className={`dropdown-autocomplete dropdown-toggle ${buttonSize} ${buttonColor}`}
+            >
+              <input
+                ref="dropdownAutoComplete"
+                className="dropdown-autocomplete--input"
+                type="text"
+                autoFocus={true}
```

Comment:
surprised the linter isn't suggesting to make this simply `autoFocus` -- try it out -- might not need to set it to `true`